ducktape/tests/runner.py

```diff
@@ -64,95 +64,120 @@ def __init__(self, cluster, session_context, session_logger, tests, port=Console
 
         self.proc_list = []
         self.staged_tests = tests
+        self.test_id_to_test_context = {t.test_id: t for t in tests}
```

Comment:
Might be easier to just reduce these maps to a single state object for each test. Not really critical now, but at a minimum `self._test_state[test_id]` is probably clearer than these verbose names and it looks like in at least one or two places we index into both of these dicts, one right after the other.